solidity/test/BancorNetwork.js

```diff
@@ -427,342 +410,366 @@ describe('BancorNetwork', () => {
-            await bancorNetwork.claimAndConvertFor(path, value, MIN_RETURN, sender2);
+            const returnAmount = await bancorNetwork.callStatic.claimAndConvertFor(
+                path,
+                value,
+                MIN_RETURN,
+                sender2.address
+            );
+            await bancorNetwork.claimAndConvertFor(path, value, MIN_RETURN, sender2.address);
 
-            const balanceAfterTransfer = await erc20Token3.balanceOf.call(sender2);
-            expect(balanceAfterTransfer).to.be.bignumber.equal(balanceBeforeTransfer.add(returnAmount));
+            const balanceAfterTransfer = await erc20Token3.balanceOf(sender2.address);
+            expect(balanceAfterTransfer).to.be.equal(balanceBeforeTransfer.add(returnAmount));
         });
 
         it('should revert when calling claimAndConvertFor without approval', async () => {
             const path = paths.ERC1.ERC3;
-            const value = new BN(1000);
+            const value = BigNumber.from(1000);
 
-            await expectRevert.unspecified(bancorNetwork.claimAndConvertFor(path, value, MIN_RETURN, sender2));
+            await expect(bancorNetwork.claimAndConvertFor(path, value, MIN_RETURN, sender2.address)).to.be.reverted;
         });
 
         it('verifies that claimAndConvert transfers the converted amount correctly', async () => {
-            const value = new BN(1000);
-            await erc20Token1.approve(bancorNetwork.address, value, { from: sender });
+            const value = BigNumber.from(1000);
+            await erc20Token1.connect(sender).approve(bancorNetwork.address, value);
 
-            const balanceBeforeTransfer = await erc20Token2.balanceOf.call(sender);
+            const balanceBeforeTransfer = await erc20Token2.balanceOf(sender.address);
 
             const path = paths.ERC1.ERC2;
-            const returnAmount = await bancorNetwork.claimAndConvert.call(path, value, MIN_RETURN);
+            const returnAmount = await bancorNetwork.callStatic.claimAndConvert(path, value, MIN_RETURN);
             await bancorNetwork.claimAndConvert(path, value, MIN_RETURN);
 
-            const balanceAfterTransfer = await erc20Token2.balanceOf.call(sender);
-            expect(balanceAfterTransfer).to.be.bignumber.equal(balanceBeforeTransfer.add(returnAmount));
+            const balanceAfterTransfer = await erc20Token2.balanceOf(sender.address);
+            expect(balanceAfterTransfer).to.be.equal(balanceBeforeTransfer.add(returnAmount));
         });
 
         it('should revert when calling claimAndConvert without approval', async () => {
             const path = paths.ERC1.ERC3;
-            const value = new BN(1000);
+            const value = BigNumber.from(1000);
 
-            await expectRevert.unspecified(bancorNetwork.claimAndConvert(path, value, MIN_RETURN));
+            await expect(bancorNetwork.claimAndConvert(path, value, MIN_RETURN)).to.be.reverted;
```

Comment:
i'd like to avoid touching too much about other stuff but it indeed seems that we could check the revert return value, creating a trello ask and assigning it to both of us